### ISSUE TITLE
Forgot password flow

### ISSUE THREAD

**Issue by TimMikeladze** [leerob/next-saas-starter#12]
At the moment it's not possible to reset a forgotten password.

I suggest a basic reset password flow by sending the user a password reset link to their email.

Maybe using resend? But should be easily swappable with other email libraries.

**Comment by leerob** [comment]
💯 I am thinking we'd use Resend and react-email for both this, as well as the team invite flow.

https://github.com/leerob/next-saas-starter/blob/afff269d7ad8d6b90f5bf2c50dbf15b2ebc285d6/app/(login)/actions.ts#L417-L420
https://github.com/resend/react-email/tree/main/examples/resend

**Comment by MJomaa** [comment]
@leerob Yes please! react-email is also very easy. Was thinking of something like this:

![image](https://github.com/user-attachments/assets/3afdf15b-96d4-461b-a24c-e7b1038de395)

**Comment by atikkk** [comment]
hey @leerob can you please help me in local setup?

**Comment by samueltesfayegari** [comment]
> hey @leerob can you please help me in local setup?

Yes tutorial Video missing 🙌

**Comment by sanjomathew09** [comment]
// Get references to the form and its elements
const form = document.getElementById('forgot-password-form');
const emailInput = document.getElementById('email');
const errorMessage = document.getElementById('error-message');

// Email validation regex
const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

// Handle form submission
form.addEventListener('submit', (e) => {
  e.preventDefault(); // Prevent the default form submission

  const email = emailInput.value.trim();

  // Validate the email input
  if (!emailRegex.test(email)) {
    errorMessage.style.display = 'block';
    return;
  }

  // Hide the error message if validation passes
  errorMessage.style.display = 'none';

  // Simulate sending a reset link (replace with actual API call)
  alert(`A password reset link has been sent to: ${email}`);
  emailInput.value = ''; // Clear the input field
});

**Comment by imkrish7** [comment]
Is anyone currently working on this?

**Comment by s04** [comment]
Bump, also interested in this.

**Comment by jonkurtis** [comment]
> Is anyone currently working on this?

just resolved merge conflicts on https://github.com/nextjs/saas-starter/pull/49/

@leerob is there anything else that needs to be addressed? Are you accepting PRs from the community?

**Comment by yttasdfghjk** [comment]
What's the current state here? Is anyone working on this?

**Comment by angelotc** [comment]
Interested

**Comment by lauaall** [comment]
Bump!

**Comment by angelotc** [comment]
Took me a couple hours to implement this using Claude Code... Ended up using @MJomaa 's suggestion of react-email and Resend, which requires another API key (added dependency).

**Comment by PrinceAdvant** [comment]
For anyone looking for a boilerplate with forgot password already implemented, check out [Kostra](https://kostra.io/) — it's a Next.js SaaS boilerplate that includes password reset flow out of the box.
